lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2238,9 +2238,8 @@ sub dependencies ($self, $children_list = undef, $parents_list = undef) {
     for my $s (@$parents_list) {
         push(@{$parents{$s->to_string}}, $s->parent_job_id);
         $has_parents = 1;
-        my $jobs = $self->result_source->schema->resultset('Jobs');
-        $parents_ok = $jobs->find($s->parent_job_id, {select => ['result']})->is_ok
-          if $is_final && $parents_ok;
+        my $parent = $self->result_source->schema->resultset('Jobs')->find($s->parent_job_id, {select => ['result']});
+        $parents_ok = defined $parent && $parent->is_ok if $is_final && $parents_ok;
```

Comment:
not sure but the `$parents_ok` seems redundant here.
IIUC $parents_ok is initialized and it is assumed `is_ok`. then for each parent you check if the result still `is_ok`. 
the variable then it should be updated only and when if $is_final.
Why does it need to check again `$parents_ok`?